Move data to server

wtharvey2 opened the pull request:
A PR to trial http requests for use in storing session data and ID numbers

wtharvey2 committed:
Enable js file for decideMain

wtharvey2 committed:
Failed attempt to implement server request/response for id number

wtharvey2 committed:
Error somewhere in server or request code

wtharvey2 commented:
@nickvazz, can you take a look at this if you have a chance? I implemented an http request in this branch that calls for an ID number when you hit submit off of the initial home page, but am having issues getting the information to the browser JS. I think I'm either failing to attach the ID number properly on the server-side or failing to call the client-side code correctly, but haven't been able to find the issue after a decent amount of digging and attempting to use axios and fetch.

The client code is here: https://github.com/wtharvey2/Decide/blob/move-data-to-server/decideMain.js#L15-L38
Server code is here: https://github.com/wtharvey2/Decide/blob/move-data-to-server/dataComm.js#L30-L35

nickvazz commented on dataComm.js:
```diff
@@ -0,0 +1,38 @@
+})
+
+// get a unique ID number
+router.get('/newID', function (req, res) {
+  // get max id number and add 1
+  maxIDNumber += 1;
+  // return the id number
+  newIDnumber = String(maxIDNumber);
+  res.json({ 'id_num': String(500)});
+})
```
Are you returning only `String(500)` here on purpose? Is it for debug?

wtharvey2 replied:
@nickvazz It's there for debug only. Normally would return newIDnumber, which should be 1001 the first time it's requested

wtharvey2 replied:
Returning it for debug only. Normally would be newIDnumber

nickvazz commented on decideMain.html:
```diff
@@ -4,10 +4,10 @@
   </head>
   <body>
-    <form id="groupForm" action="decideQuestion" method="get">
+    <form id="groupForm" onsubmit="makeNewURL()" method="post">
```
This looks like its a post where other stuff is get requests?

wtharvey2 replied:
Crud. I had updated this, but hadn't pushed to the remote. I'll push now. It will have a different attempt (using axios I think)

nickvazz commented:
Looks like it might be a post/get request issue as a first pass. How would I build locally, there's a `dockerfile` now, is it all set for production but what about local debug? (or how are _you_ locally debugging 😛)

wtharvey2 committed:
More attempts to fix repo

wtharvey2 commented:
@nickvazz, I'm debugging just using a node server locally. I'm deploying it in docker on a [forked branch. 
.](https://github.com/wtharvey2-docker/Decide)

wtharvey2 committed:
Removed attempt to implement axios

wtharvey2 commented:
I don't think I'm successfully sending anything in the body of this response based on a few print statements that I put in the server side (and have since removed). https://github.com/wtharvey2/Decide/blob/move-data-to-server/dataComm.js#L5-L14

wtharvey2 committed:
HTTP request for ID number is now synchronous

- This is not ideal and needs to be ammended as a TODO
- For now it works though because this must be done to set all the form data to then send for use in the question page

wtharvey2 committed:
Added ID Value to form before its submitted

wtharvey2 committed:
Added debug prints

wtharvey2 committed:
Changed form submit to get and drafted means of posting form data

- Tried turning form data into a json to store on the server. However, the json string is not being sent in the http request

wtharvey2 committed:
Question posting working

Need to set up a database for better data longevity, but dictionary implementation will work for now

wtharvey2 committed:
Initial URL parse works

wtharvey2 committed:
Fixed a few typos that prevented function

wtharvey2 committed:
Ideas are now handled server-side

Functional for one instance to submit and another to receive it on reload

wtharvey2 committed:
Code to prep voting screen and add votes

wtharvey2 committed:
Successful load of result screen. Failure to calculate winner

wtharvey2 committed:
Basic Functionality appears to work

wtharvey2 committed:
All buttons shown should be functional now

- Minimal testing has been completed
- Many bugs may still exist
- Need to write test cases
